keps/sig-api-machinery/0006-apply.md

```diff
@@ -299,6 +301,77 @@ The conversion between the two and creating the diff was complex and would have
 
 - 12/2019 [#86083](https://github.com/kubernetes/kubernetes/pull/86083) implementing a poc for the described approach
 
+#### Fix Admission Chain for Subresources
+
+##### Current Behavior
+
+Currently, the admission chain on a subresource is not propagated for the parent
+resource.
+
+When a user makes a request to a subresource, then admission webhooks and
+validation are skipped on the parent, and we don't have the user information to
+set the field manager for the parent.
+
+For example, a validating webhook on Deployments to enforce a replica policy
+can be bypassed by any /scale subresource request because
+the webhook admission chain is not called on the subresource. We cannot track
+that the user who made the request to the /scale resource owns the replicas
+field.
+
+##### Proposed Change
+
+The webhooks will be called for the subresource and parent using the same
+admission context, such that the original user is represented in the admission
+attributes. Webhooks will opt-in to this behavior on subresources in the
+registration.
+
+The webhooks will be called in the following order, considering a request to
+`/foo/scale` as an example:
+1. mutating webhooks will be called on the included subresource (/foo/scale)
+2. mutating webhooks will be called on the parent (/foo)
+3. optionally, mutating webhooks will be called a second time in the same order
+   (1), then (2) (/foo/scale, then /foo)
+4. validating webhooks will be called in parallel for the subresource and parent
+   (/foo/scale and /foo simultaneously)
+
+Note that when applying mutating webhooks on the included subresources (1),
+then any modification attempts to implicitly modify fields of the parent that
+are not explicitly fields of the subresource will be discarded before writing to
+storage.
+
+After webhooks and validation pass, then we'll use the user in the
+admission context to properly track the field manager for server-side apply when
+writing the parent to storage.
+
+We'll add the new field `includeSubresources` to the admission webhook
```

Comment:
Maybe mention that this is for compatibility reasons

Reply:
Done.